Drop Python 2.7 and 3.5 support and bump dependency upper bounds

RobbeSneyders opened the pull request:
Fixes #130
Fixes #135 

`jsonschema` released version 4.0.0 which dropped support for Python 2.7 and 3.5. `openapi-spec-validator` should do the same to support new versions of `jsonschema`. Both upstream dependencies and downstream dependents have already upgraded their `jsonschema` version constraints, which leads to conflicts with `openapi-spec-validator`.

sthagen reviewed (approved):
LGTM … whatever you people make out of that 🙈

p1c2u commented on pyproject.toml:
```diff
@@ -34,21 +32,20 @@ classifiers = [
-six = "*"
-pathlib2 = {version = "*", python = "~2.7"}
-openapi-schema-validator = "<0.2.0"
+pathlib2 = {version = "*"}
```
python3 already has `pathlib` so this can be removed also?

p1c2u commented on pyproject.toml:
```diff
@@ -34,21 +32,20 @@ classifiers = [
 
 [tool.poetry.dev-dependencies]
-mock = {version = "*", python = "~2.7"}
+mock = {version = "*"}
```
we should use python3's `unites.mock`

nezhar commented:
@p1c2u I would also suggest to drop support for Python 3.6 and to add suport for Python 3.10 acording to https://endoflife.date/python

RobbeSneyders commented:
Good catch @p1c2u, addressed your comments.
I also dropped python 3.6 in favor of 3.10 as mentioned by @nezhar. This is in line with `jsonschema` as well.

nezhar commented on .github/workflows/python-test.yml:
```diff
@@ -14,7 +14,7 @@ jobs:
     strategy:
       matrix:
-        python-version: [2.7, 3.5, 3.6, 3.7, 3.8, 3.9]
+        python-version: [3.7, 3.8, 3.9, 3.10]
```
The Python versions should be declared as strings as in the [docs of the setup python action](https://github.com/actions/setup-python#usage), otherwise `3.10` might be interpreted as `3.1` in the workflow.

p1c2u commented:
@RobbeSneyders thank you for the contribution
